tests/conftest.py

```diff
@@ -731,18 +712,37 @@ def default_execution_context() -> ExecutionContext:
     return ExecutionContext(GraphSchema({}), uuid.uuid4().hex)
 
 
-# TODO: fix this
+@pytest.fixture(scope="session", autouse=True)
+def temp_cache_for_fixtures(tmp_path_factory: TempPathFactory) -> None:
+    # This fixture makes sure that wide fixtures which don't have `function` scope
+    # (session, package, module) don't use the global
+    # cache. If you want to use the cache in a session scoped fixture, then please
+    # consider using the `enable_cache` context manager.
+    old_get_cache_location = LocalTrainingCache._get_cache_location
+    LocalTrainingCache._get_cache_location = lambda: tmp_path_factory.mktemp(
+        f"cache-{uuid.uuid4()}"
+    )
+
+    yield
+
+    LocalTrainingCache._get_cache_location = old_get_cache_location
+
+
 @pytest.fixture(autouse=True)
 def use_temp_dir_for_cache(
     monkeypatch: MonkeyPatch, tmp_path_factory: TempdirFactory
-) -> Path:
-    # This avoids that the unit tests because they share a local cache.
-    # Instead we are creating a new cache for every test.
+) -> None:
+    # This fixture makes sure that a single test function has a constant cache
+    # cache.
     cache_dir = tmp_path_factory.mktemp(uuid.uuid4().hex)
-    monkeypatch.setattr(
-        LocalTrainingCache,
-        LocalTrainingCache._get_cache_location.__name__,
-        Mock(return_value=cache_dir),
-    )
+    monkeypatch.setattr(LocalTrainingCache, "_get_cache_location", lambda: cache_dir)
```

Comment:
I assume this has no effect if the cache has already been created in a fixture within `enable_cache`?

Reply:
yep - `enable_cache` would just override it. 

With the two fixtures we basically get the best out of two worlds: No global cache for `session_wide` things + a working function scoped cache